zipp/__init__.py

```diff
@@ -313,6 +327,38 @@ def iterdir(self):
         subs = map(self._next, self.root.namelist())
         return filter(self._is_child, subs)
 
+    def match(self, path_pattern):
+        return pathlib.Path(self.at).match(path_pattern)
```

Comment:
I'm yet unsure if `pathlib.Path` is the right behavior here. I think we should investigate what `match` is meant to take as input and see how that applies to zipp./zipfile.Path objects.

Reply:
I'm not going to worry about this concern for now. I'd like to get this implementation out so others can try it out and maybe get some feedback before refining the behavior further.